sync3/handler/connstate_live.go

```diff
@@ -444,3 +444,20 @@ func (s *connStateLive) resort(
 	}
 	return ops, hasUpdates
 }
+
+// shouldIncludeHeroes returns whether the given roomID is in a list or direct
+// subscription which should return heroes.
+func (s *connStateLive) shouldIncludeHeroes(roomID string) bool {
+	roomIDsToLists := s.lists.ListsByVisibleRoomIDs(s.muxedReq.Lists)
+	subscriptionInclude := s.roomSubscriptions[roomID].IncludeHeroes()
```

Comment:
Rejig this so it's faster. Checking the lists is expensive.
```go
if s.roomSubscriptions[roomID].IncludeHeroes() {
    return true
}
// rest of list checks
```

Reply:
Right, that's easier.